vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,539 @@
+  reframe(
+    reference_date = reference_date,
+    report_date = seq(reference_date, min(
+      reference_date + max_delay,
+      nowcast_date
+    ),
+    by = "day"
+    )
+  ) |>
+  mutate(delay = as.integer(report_date - reference_date)) |>
+  left_join(training_df,
+    by = c("reference_date", "report_date")
+  ) |>
+  replace_na(list(count = 0)) |>
+  select(-report_date) |>
+  pivot_wider(
+    names_from = delay,
+    values_from = count
+  )
+head(rep_tri_df)
+```
+
+As a final pre-processing step, we'll format the reporting triangle as just a matrix, for compatibility with the low-level functions in `baselinenowcast`.
+```{r}
+rep_tri <- rep_tri_df |>
+  select(-reference_date) |>
+  as.matrix()
+```
+
+## Specify the `baselinenowcast` model
+
+We will use the helper function `allocate_reference_times()` to determine how many reference times to use for delay estimation and uncertainty estimation, using the package defaults.
+See the "Default Settings" section of the [mathematical model](model_definition.html) vignette for more details.
+```{r}
+ref_time_allocation <- allocate_reference_times(rep_tri)
+n_history_delay <- ref_time_allocation$n_history_delay
+n_retrospective_nowcasts <- ref_time_allocation$n_retrospective_nowcasts
+```
+
+# Run the `baselinenowcast` workflow
+
+First, we generate the point estimate of the nowcast as a "filled in" reporting triangle.
+See the sections "Estimating the delay distribution from a reporting triangle" and "Point nowcast generation" of the [mathematical model](model_definition.html) for more details.
+```{r}
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
+```
+
+Next, we estimate and apply uncertainty by generating retrospective nowcasts and comparing against later observed data.
+See the section "Uncertainty quantification" of the [mathematical_model](model_definition.html) for more details on assumptions being made here and a further description of how the uncertainty is quantified using past nowcast errors.
+```{r}
+nowcast_draws_df <- estimate_and_apply_uncertainty(
+  pt_nowcast_matrix,
+  rep_tri,
+  n_history_delay = n_history_delay,
+  n_retrospective_nowcasts = n_retrospective_nowcasts
+)
+
+head(nowcast_draws_df)
+```
+
+# Summarise and plot the nowcast
+
+Now that we have a dataframe of probabilistic nowcast draws, which represent our predicted "final" case count at each reference time, we can summarise them to compute the 50th and 95th percent prediction intervals, which are used to communicate uncertainty in the nowcast.
+```{r}
+nowcast_summary_df <-
+  nowcast_draws_df |>
+  group_by(time) |>
+  summarise(
+    median = median(pred_count),
+    q50th_lb = quantile(pred_count, 0.25),
+    q50th_ub = quantile(pred_count, 0.75),
+    q95th_lb = quantile(pred_count, 0.025),
+    q95th_ub = quantile(pred_count, 0.975)
+  )
+```
+
+Next, we will join the initially reported cases and the held out, true final cases to our probabilistic nowcast.
```

Comment:
Why? Same points. 

We need to do X therefore we will do why. No one cares about technical detail for its own sake we need to justify every step first.